src/v/utils/fragmented_vector.h

```diff
@@ -36,15 +36,42 @@
  * all of the space at once we might benefit from the allocator helping with
  * giving us contiguous memory.
  */
-template<typename T, size_t fragment_size = 8192>
+template<typename T, size_t max_fragment_size = 8192>
 class fragmented_vector {
-    static constexpr size_t elems_per_frag = fragment_size / sizeof(T);
+    // calculate the maximum number of elements per fragment while
+    // keeping the element count a power of two
+    static constexpr size_t calc_elems_per_frag(size_t esize) {
+        size_t max = max_fragment_size / esize;
+        assert(max > 0);
+        // round down to a power of two
+        size_t pow2 = 1;
+        while (pow2 * 2 <= max) {
+            pow2 *= 2;
+        }
+        return pow2;
+    }
+
+    static constexpr size_t elems_per_frag = calc_elems_per_frag(sizeof(T));
+    static constexpr size_t frag_bytes = elems_per_frag * sizeof(T);
```

Comment:
this second one looks unused (at least in this commit)

Reply:
Removed in latest push.